### ISSUE TITLE
Forbid expression that whole expression is a group

### ISSUE THREAD

**Issue by fisker** [ota-meshi/eslint-plugin-regexp#814]
**Motivation**

I can't find a case that make sense to use `/(.)/` if used in `String.match`, `match[0]` can be used instead of `match[1]`, if want a substitution, `$&` can be used instead of `$1`. Maybe I was wrong about this, but there are certainly cases we can avoid group.

**Description**

See above

**Examples**

```js
/* ✓ GOOD */
var [matched] = foo.match(/a/);
foo.replace(/a/, '-$&');

/* ✗ BAD */
var [, matched] = foo.match(/(a)/);
foo.replace(/(a)/, '-$1');
```
